src/Acuminator/Acuminator.Utilities/Roslyn/Semantic/PXGraph/PXGraphSemanticModel.cs

```diff
@@ -0,0 +1,195 @@
+﻿using Acuminator.Utilities.Common;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+
+namespace Acuminator.Utilities.Roslyn.Semantic.PXGraph
+{
+    public class PXGraphSemanticModel
+    {
+        private readonly CancellationToken _cancellation;
+        private readonly PXContext _pxContext;
+
+        public GraphType Type { get; private set; }
+        public INamedTypeSymbol Symbol { get; private set; }
+        public (ConstructorDeclarationSyntax Node, IMethodSymbol Symbol) StaticCtrInfo { get; private set; }
+        public ImmutableArray<GraphInitializerInfo> Initializers { get; private set; }
+
+        private PXGraphSemanticModel(PXContext pxContext, GraphType type, INamedTypeSymbol symbol, CancellationToken cancellation = default)
+        {
+            cancellation.ThrowIfCancellationRequested();
+
+            _cancellation = cancellation;
+            _pxContext = pxContext;
+            Type = type;
+            Symbol = symbol;
+
+            InitStaticCtr();
+            InitDeclaredInitializers();
+        }
+
+        private void InitStaticCtr()
```

Comment:
@metallmonster this method could be inlined and the `StaticCtrInfo` made readonly

Reply:
@SENya1990 compiler is smart enough to inline most of the methods by itself, let's not spend time on unneeded micro-optimizations.

Reply:
@dropsonic it's not about inlining, it's more about allowing the  StaticCtrInfo to be readonly

Reply:
@SENya1990 what is the reason to inline it? It is called only from one place.
Yes, but in this case `StaticCtrInfo` would require additional backing field which is more complex than current implementation

Reply:
@metallmonster no it won't:
```
 public (ConstructorDeclarationSyntax Node, IMethodSymbol Symbol) StaticCtrInfo { get; }

private PXGraphSemanticModel(PXContext pxContext, GraphType type, INamedTypeSymbol symbol, CancellationToken cancellation = default)
        {
 cancellation.ThrowIfCancellationRequested();
            _cancellation = cancellation;
            _pxContext = pxContext;
            Type = type;
            Symbol = symbol;
            StaticCtrInfo = Symbol.GetDeclaredStaticConstructor(_cancellation);
            InitDeclaredInitializers();
}

```